frame/babe/src/lib.rs

```diff
@@ -224,10 +225,16 @@ decl_storage! {
 		/// on block finalization. Querying this storage entry outside of block
 		/// execution context should always yield zero.
 		Lateness get(fn lateness): T::BlockNumber;
+
+		EpochConfig: BabeEpochConfiguration;
+		NextEpochConfig: Option<BabeEpochConfiguration>;
 	}
 	add_extra_genesis {
 		config(authorities): Vec<(AuthorityId, BabeAuthorityWeight)>;
-		build(|config| Module::<T>::initialize_authorities(&config.authorities))
+		config(epoch_config): BabeEpochConfiguration;
+		build(|config| Module::<T>::initialize_authorities_and_epoch_config(
+			&config.authorities, &config.epoch_config
```

Comment:
We should just initialize `EpochConfig` here (i.e. `EpochConfig::put(config.epoch_config`), instead of changing the function to initialize authorities. As you've seen below, authorities can be initialized either by genesis config or by the session module (in which case the authorities vec here will be empty), which would lead to `EpochConfig` being erroneously initialized with a default value.